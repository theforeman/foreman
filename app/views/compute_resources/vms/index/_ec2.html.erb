<table class="list">
  <tr>
    <thead>
      <th>Name</th>
      <th>DNS</th>
      <th>Type</th>
      <th>State</th>
      <th></th>
    </thead>
  </tr>
  <% @compute_resource.vms.each do |vm| -%>
    <tr>
      <td><%= vm.tags["Name"] %></td>
      <td><%= vm.dns_name %></td>
      <td><%= vm.flavor_id %></td>
      <td <%= vm_power_class(vm.ready?)%>> <%= vm_state(!vm.ready?) %> </td>
      <td>
        <%= vm_power_action vm %>
        <%= display_link_if_authorized "Delete", hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.id).merge(:auth_action => :destroy), :confirm => 'Are you sure?', :method => :delete, :class => "label important" %>
      </td>
    </tr>
  <% end -%>
</table>
