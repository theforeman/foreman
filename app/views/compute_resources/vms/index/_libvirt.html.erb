<table class="list">
  <tr>
    <thead>
      <th> Name </th>
      <th> CPUs </th>
      <th> Memory </th>
      <th> Power </th>
      <th></th>
    </thead>
  </tr>
  <% @compute_resource.vms.each do |vm| -%>
    <tr>
      <td><%= link_to vm.name, hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.uuid) %></td>
      <td><%= vm.cpus %></td>
      <td> <%= number_to_human_size vm.memory_size*1024 %> </td>
      <td <%= vm_power_class(vm.ready?)%>> <%= vm_state(!vm.ready?) %> </td>
      <td>
        <%= vm_power_action vm %>
        <%= display_link_if_authorized "Delete", hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.uuid).merge(:auth_action => :destroy), :confirm => 'Are you sure?', :method => :delete, :class => "label important" %>
      </td>
    </tr>
  <% end -%>
</table>
