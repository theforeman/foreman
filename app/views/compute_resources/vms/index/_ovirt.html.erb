<table class="list">
  <tr>
    <thead>
    <th>Name</th>
    <th> CPUs</th>
    <th> Memory</th>
    <th> Power</th>
    <th></th>
    </thead>
  </tr>
  <% @compute_resource.vms.each do |vm| -%>
    <tr>
      <td><%= vm.name %></td>
      <td><%= vm.cores %></td>
      <td> <%= number_to_human_size vm.memory %> </td>
      <td <%= vm_power_class(vm.ready?) %>> <%= vm_state(!vm.ready?) %> </td>
      <td>
        <%= vm_power_action vm %>
        <%= display_link_if_authorized "Delete", hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.id).merge(:auth_action => :destroy), :confirm => 'Are you sure?', :method => :delete, :class => "label important" %>
      </td>

    </tr>
  <% end -%>
</table>


