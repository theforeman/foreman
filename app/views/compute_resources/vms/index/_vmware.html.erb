<table class="table table-bordered table-striped">
  <tr>
    <thead>
    <th>Name</th>
    <th> CPUs</th>
    <th> Memory</th>
    <th> Power</th>
    <th></th>
    </thead>
  </tr>
  <% @compute_resource.vms.each do |vm| -%>
    <tr>
      <td><%= vm.name %></td>
      <td><%= vm.cpus %></td>
      <td> <%= number_to_human_size vm.memory %></td>
      <td <%= vm_power_class(vm.ready?) %>> <%= vm_state(!vm.ready?) %> </td>
      <td>
        <%= vm_power_action vm %>
        <%= display_link_if_authorized "Console", hash_for_console_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.id), :class => "label label-info" %>
        <%= display_link_if_authorized "Delete", hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.id).merge(:auth_action => :destroy), :confirm => 'Are you sure?', :method => :delete, :class => "label label-important" %>
      </td>

    </tr>
  <% end -%>
</table>
