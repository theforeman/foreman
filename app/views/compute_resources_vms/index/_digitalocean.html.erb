<table class="table table-bordered" data-table='inline'>
  <thead>
    <tr>
      <th><%= _('Name') %></th>
      <th><%= _('Image') %></th>
      <th><%= _('Type') %></th>
      <th><%= _('Region') %></th>
      <th><%= _('State') %></th>
      <th></th>
    </tr>
  </thead>
  <% @vms.each do |vm| %>
    <tr>
      <td><%= link_to_if_authorized vm.name, hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.identity).merge(:auth_object => @compute_resource, :authorizer => authorizer) %></td>
      <td><%= vm.image.name %></td>
      <td><%= vm.flavor.name %></td>
      <td><%= vm.region.name %></td>
      <td> <span <%= vm_power_class(vm.ready?) %>> <%= vm_state(vm) %></span> </td>
      <td>
        <%= action_buttons(
                vm_power_action(vm, authorizer),
                display_delete_if_authorized(hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.id).merge(:auth_object => @compute_resource, :authorizer => authorizer))) %>
      </td>
    </tr>
  <% end %>
</table>
