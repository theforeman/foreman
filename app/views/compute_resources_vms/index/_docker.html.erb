<table class="table table-bordered" data-table='inline'>
  <thead>
    <tr>
      <th><%= _('Name')   %></th>
      <th><%= _('Image')  %></th>
      <th><%= _('CPUs')   %></th>
      <th><%= _('Memory') %></th>
      <th><%= _('Status') %></th>
      <th><%= _('Power')  %></th>
      <th></th>
    </tr>
  </thead>
  <% @vms.each do |vm| %>
    <tr>
      <% name = vm.name || (!vm.attributes['names'].empty? && vm.attributes['names'].first) %>
      <td><%= link_to_if_authorized name, hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.identity) %></td>
      <td><%= vm.image %></td>
      <td><%= vm.cores %></td>
      <td> <%= number_to_human_size vm.memory %> </td>
      <td> <%= vm.attributes['status'] %> </td>
      <td> <span <%= vm_power_class(vm.ready?) %>> <%= vm_state(vm) %></span> </td>
      <td>
        <%= action_buttons(vm_power_action(vm),
                           display_delete_if_authorized(hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.id))) %>
      </td>

    </tr>
  <% end %>
</table>
