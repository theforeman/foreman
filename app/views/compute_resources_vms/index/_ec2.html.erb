<table class="table table-bordered" data-table="inline">
  <thead>
    <tr>
      <th><%= _('Name') %></th>
      <th><%= _('DNS') %></th>
      <th><%= _('Type') %></th>
      <th><%= _('State') %></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <% @vms.each do |vm| %>
      <tr>
        <td><%= link_to_if_authorized vm.name, hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.identity).merge(:auth_object => @compute_resource, :authorizer => authorizer) %></td>
        <td><%= vm.dns %></td>
        <td><%= vm.flavor_id %></td>
        <td>
          <span <%= vm_power_class(vm.ready?) %>>
            <%= vm_state(vm) %></span>
        </td>
        <td>
          <%= unless vm.state == 'terminated' or vm.state == 'pending'
                action_buttons(vm_power_action(vm, authorizer),
                               display_delete_if_authorized(hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.id).merge(:auth_object => @compute_resource, :authorizer => authorizer)))
              end
          %>
        </td>
      </tr>
    <% end %>
  </tbody>
</table>
