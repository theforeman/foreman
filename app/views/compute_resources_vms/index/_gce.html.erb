<thead>
  <tr>
    <th><%= _('Name') %></th>
    <th><%= _('Type') %></th>
    <th><%= _('State') %></th>
    <th><%= _('Actions') %></th>
  </tr>
</thead>
<tbody>
  <% @vms.each do |vm| %>
    <tr>
      <td><%= link_to_if_authorized vm.name, hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.identity).merge(:auth_object => @compute_resource, :auth_action => 'view', :authorizer => authorizer) %></td>
      <td><%= vm.pretty_machine_type %></td>
      <td><%= vm.status.downcase %></td>
      <td>
        <%= action_buttons(vm_power_action(vm, authorizer),
                           vm_import_action(vm),
                           display_delete_if_authorized(hash_for_compute_resource_vm_path(:compute_resource_id => @compute_resource, :id => vm.identity).merge(:auth_object => @compute_resource, :authorizer => authorizer))) %>
      </td>
    </tr>
  <% end %>
</tbody>
