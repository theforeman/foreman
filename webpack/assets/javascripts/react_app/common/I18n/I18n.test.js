import Jed from 'jed';
import { translate, ngettext } from './index';

jest.unmock('./index');
jest.unmock('jed');

describe('gettext', () => {
  Jed.gettext = jest.fn(s => s);
  it('chevrons should not be presented', () => {
    expect(translate('should not be with chevrons')).toMatchSnapshot();
  });
  it('chevrons should be presented', () => {
    global.I18N_MARK = true;
    expect(translate('should be with chevrons')).toMatchSnapshot();
  });
});

describe('ngettext', () => {
  Jed.ngettext = jest.fn(s => s);
  it('chevrons should not be presented', () => {
    expect(ngettext('should not be with chevrons')).toMatchSnapshot();
  });
  it('chevrons should be presented', () => {
    global.I18N_MARK = true;
    expect(ngettext('should be with chevrons')).toMatchSnapshot();
  });
});
