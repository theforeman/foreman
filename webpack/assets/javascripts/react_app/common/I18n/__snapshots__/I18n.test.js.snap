// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gettext chevrons should be presented 1`] = `"»should be with chevrons«"`;

exports[`gettext chevrons should not be presented 1`] = `"should not be with chevrons"`;

exports[`ngettext chevrons should be presented 1`] = `"»should be with chevrons«"`;

exports[`ngettext chevrons should not be presented 1`] = `"»should not be with chevrons«"`;
